### PR TITLE
feat(schemas)[OK-63] Add SlideObject schema

### PR DESCRIPTION
**Pull request by Wajktor13**

### CODE REVIEW

File: backend/src/schemas/slide-object.schema.ts [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+  color: string;
+  blur: number;
+  offsetX: number;
+  offsetY: number;
+}
+
+@Schema({ timestamps: true })
+export class SlideObject {
+
+  // relations
```
Comment (marcepanowyy) [inline backend/src/schemas/slide-object.schema.ts]: I would suggest moving the relations to the very end, as it is for the other schemas

File: backend/src/schemas/slide.schema.ts [hunk 2]
```diff
@@ -1,82 +1,7 @@
...
-  cropX?: number;
-  cropY?: number;
-  filters?: {
-    type: string;
-    value: number;
-  }[];
-}
 
 @Schema({ timestamps: true })
 export class Slide {
```
Comment (marcepanowyy) [inline backend/src/schemas/slide.schema.ts:12]: The are no real relationships between [Slide & SlideObject](https://github.com/OBELISK-TEAM/OBELISK/blob/13c89eaf6763740340942541c2a997102846b473/backend/src/schemas/slide.schema.ts#L15) (one to many) and [Slide & Board](https://github.com/OBELISK-TEAM/OBELISK/blob/13c89eaf6763740340942541c2a997102846b473/backend/src/schemas/slide.schema.ts#L25) (many to one). Take a look at other relationships like this one between [User & Board](https://github.com/OBELISK-TEAM/OBELISK/blob/13c89eaf6763740340942541c2a997102846b473/backend/src/schemas/user.schema.ts#L45) and [Board & User](https://github.com/OBELISK-TEAM/OBELISK/blob/13c89eaf6763740340942541c2a997102846b473/backend/src/schemas/board.schema.ts#L50)

**Review by marcepanowyy — CHANGES_REQUESTED** [review]

File: backend/src/modules/slides/slides.dto.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { IsNotEmpty, IsString } from 'class-validator';
```
Comment (marcepanowyy) [inline backend/src/modules/slides/slides.dto.ts:1]:
```
import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
import { SlideObject } from '../../schemas/slide-object.schema';
import {
  Schema as MongooseSchema,
  Document as MongooseDocument,
} from 'mongoose';
```
for uniform import names

File: backend/src/modules/slides/slides.dto.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
...
-import { SlideObject } from '../../schemas/slide.schema';
+import { SlideObject } from '../../schemas/slide-object.schema';
 
 export class CreateSlideDto {
```
Comment (marcepanowyy) [inline backend/src/modules/slides/slides.dto.ts:4]: Dto should be in slide module

File: backend/src/schemas/slide-object.schema.ts [hunk 5]
```diff
@@ -0,0 +1,227 @@
+import mongoose from 'mongoose';
```
Comment (marcepanowyy) [inline backend/src/schemas/slide-object.schema.ts]: Uniform imports just like above

File: backend/src/schemas/slide-object.schema.ts [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+    type: [{ type: mongoose.Schema.Types.Mixed }],
+  })
+  filters?: {
+    type: string;
+    value: number;
+  }[];
+
+  // relations
+
+  @Prop({ required: true, type: mongoose.Schema.Types.ObjectId, ref: 'User' })
```
Comment (marcepanowyy) [inline backend/src/schemas/slide-object.schema.ts]: code reformat

File: backend/src/schemas/slide-object.schema.ts [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+  @Prop({ type: String })
+  src?: string;
+
+  @Prop({ type: Number })
+  cropX?: number;
+
+  @Prop({ type: Number })
+  cropY?: number;
+
+  @Prop({
```
Comment (marcepanowyy) [inline backend/src/schemas/slide-object.schema.ts:334]: code reformat
image

File: backend/src/schemas/slide-object.schema.ts [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+    type: string;
+    value: number;
+  }[];
+
+  // relations
+
+  @Prop({ required: true, type: mongoose.Schema.Types.ObjectId, ref: 'User' })
+  createdBy: User;
+
+  @Prop({ required: true, type: mongoose.Schema.Types.ObjectId, ref: 'Slide'})
```
Comment (marcepanowyy) [inline backend/src/schemas/slide-object.schema.ts]: code reformat

**Review by marcepanowyy — CHANGES_REQUESTED** [review]
Besides commented issues, I would suggest implementing a base controller and service to test out whether the relations are working well

**Comment by marcepanowyy** [comment]
Please change PR name to fit into the pattern and update the branch

File: backend/src/modules/slides/slides.dto.ts (hunk 3 above)
Comment (Wajktor13) [reply backend/src/modules/slides/slides.dto.ts:1]: Not sure what you want to change here

File: backend/src/modules/slides/slides.dto.ts (hunk 4 above)
Comment (Wajktor13) [reply backend/src/modules/slides/slides.dto.ts:4]: You mean SlideDto should use SlideObjectDto? Could you elaborate?

File: backend/src/schemas/slide-object.schema.ts (hunk 7 above)
Comment (Wajktor13) [reply backend/src/schemas/slide-object.schema.ts:334]: Previous version was more readable for me, besides we wen from 230 to 360 lines. But whatever.

**Comment by Wajktor13** [comment]
> Besides commented issues, I would suggest implementing a base controller and service to test out whether the relations are working well

Not in scope of this MR

File: backend/src/modules/slides/slides.dto.ts (hunk 3 above)
Comment (marcepanowyy) [reply backend/src/modules/slides/slides.dto.ts:1]: my mistake, thought it was schema file

File: backend/src/schemas/slide.schema.ts [hunk 9]
```diff
@@ -1,82 +1,8 @@
+import { Schema as MongooseSchema } from 'mongoose';
 import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
 import { Board } from './board.schema';
+import { SlideObject } from './slide-object.schema';
```
Comment (marcepanowyy) [inline backend/src/schemas/slide.schema.ts:8]:
```typescript
export type SlideDocument = Slide & MongooseDocument
```

**Review by marcepanowyy — CHANGES_REQUESTED** [review]

**Comment by marcepanowyy** [comment]
> > Besides commented issues, I would suggest implementing a base controller and service to test out whether the relations are working well
> 
> Not in scope of this MR

Yeah, but I would not move the ticket to done section until the base endpoint are implemented to test whether the code is working or not

File: backend/src/schemas/slide-object.schema.ts (hunk 6 above)
Comment (Wajktor13) [reply backend/src/schemas/slide-object.schema.ts]: Could you elaborate?

File: backend/src/schemas/slide-object.schema.ts (hunk 8 above)
Comment (Wajktor13) [reply backend/src/schemas/slide-object.schema.ts]: Could you elaborate?

File: backend/src/schemas/slide.schema.ts (hunk 9 above)
Comment (Wajktor13) [reply backend/src/schemas/slide.schema.ts:8]: Not really in scope of this PR, but done.

File: backend/src/schemas/slide-object.schema.ts [hunk 10]
```diff
@@ -0,0 +1,365 @@
...
+import { User } from './user.schema';
+import { Slide } from './slide.schema';
+import {
+  Schema as MongooseSchema,
+  Document as MongooseDocument,
+} from 'mongoose';
+
+export type SlideObjectDocument = SlideObject & MongooseDocument;
+
+export class Shadow {
```
Comment (marcepanowyy) [inline backend/src/schemas/slide-object.schema.ts]: there should be interface located in shared/interfaces

File: backend/src/schemas/slide-object.schema.ts [hunk 11]
```diff
@@ -0,0 +1,365 @@
...
+
+  @Prop({
+    type: Number,
+  })
+  cropY?: number;
+
+  @Prop({
+    type: [{ type: MongooseSchema.Types.Mixed }],
+  })
+  filters?: {
```
Comment (marcepanowyy) [inline backend/src/schemas/slide-object.schema.ts:337]: export interface Filter in shared/interfaces

**Review by marcepanowyy — CHANGES_REQUESTED** [review]

**Review by marcepanowyy — APPROVED** [review]

**Pull request merged**